widgetssdk/src/main/java/com/glia/widgets/survey/SurveyAdapter.java

```diff
@@ -67,84 +85,16 @@ private static View createNewLayout(ViewGroup parent, @LayoutRes int layoutId) {
-            bindOpenText(question, answer, (SurveyOpenTextViewHolder) viewHolder);
-        }
-    }
-
-    private void bindScale(QuestionItem questionItem, Survey.Question question, Survey.Answer answer, SurveyScaleViewHolder scaleViewHolder) {
-        setItemTitle(scaleViewHolder.titleTextView, question.getText(), question.isRequired());
-
-        scaleViewHolder.setAnswer(answer);
-
-        scaleViewHolder.listener = value -> {
-            setAnswer(questionItem, value);
-            scaleViewHolder.setSelected(value);
-        };
-    }
-
-    private void bindSingle(QuestionItem questionItem, Survey.Question question, SurveySingleChoiceViewHolder singleChoiceViewHolder) {
-        setItemTitle(singleChoiceViewHolder.title, question.getText(), question.isRequired());
-
-        singleChoiceViewHolder.singleChoice(question.getOptions(), questionItem);
-    }
-
-    private void bindYesNo(QuestionItem questionItem, Survey.Question question, Survey.Answer answer, SurveyYesNoViewHolder yesNoViewHolder) {
-        setItemTitle(yesNoViewHolder.titleTextView, question.getText(), question.isRequired());
-
-        yesNoViewHolder.setAnswer(answer);
-
-        yesNoViewHolder.listener = value -> {
-            setAnswer(questionItem, value);
-            yesNoViewHolder.setSelected(value);
-        };
-    }
-
-    private void bindOpenText(Survey.Question question, Survey.Answer answer, SurveyOpenTextViewHolder surveyOpenTextViewHolder) {
-        setItemTitle(surveyOpenTextViewHolder.titleComment, question.getText(), question.isRequired());
-        EditText comment = surveyOpenTextViewHolder.comment;
-        if (answer != null) {
-            comment.setText(answer.getResponse());
-        }
-    }
-
-    private void setItemTitle(TextView textView, String title, boolean require) {
-        if (require) {
-            Context context = textView.getContext();
-            int color = ContextCompat.getColor(context, R.color.glia_system_negative_color);
-            String colorString = String.format("%X", color).substring(2);
-            String source = context.getString(R.string.glia_survey_require_label, title, colorString);
-            textView.setText(Html.fromHtml(source, Html.FROM_HTML_MODE_LEGACY));
-        } else {
-            textView.setText(title);
-        }
-    }
-
-    private void setAnswer(QuestionItem questionItem, int response) {
-        questionItem.setAnswer(Survey.Answer.makeAnswer(questionItem.getQuestion().getId(), response));
-    }
-
-    private void setAnswer(QuestionItem questionItem, boolean response) {
-        questionItem.setAnswer(Survey.Answer.makeAnswer(questionItem.getQuestion().getId(), response));
-    }
-
-    private void setAnswer(QuestionItem questionItem, String response) {
-        questionItem.setAnswer(Survey.Answer.makeAnswer(questionItem.getQuestion().getId(), response));
+        ((SurveyViewHolder) viewHolder).onBind(questionItem, question, answer, listener);
```

Comment:
seems no point to cache question and answer to a separate vars since the whole questionItem is already passed here.